lib/api/apiUtils/object/versioning.js

```diff
@@ -55,14 +55,46 @@ function _storeNullVersionMD(bucketName, objKey, objMD, options, log, cb) {
     });
 }
 
+function _getVersionLocation(bucketName, objKey, options, log, cb) {
+    return metadata.getObjectMD(bucketName, objKey, options, log,
+        (err, versionMD) => {
+            if (err) {
+                log.debug('err from metadata getting specified version', {
+                    error: err,
+                    method: '_getVersionDataLocation',
+                });
+                return cb(err);
+            }
+            if (!versionMD.location) {
+                return cb();
+            }
+            const dataToDelete = Array.isArray(versionMD.location) ?
+                versionMD.location : [versionMD.location];
+            return cb(null, dataToDelete);
+        });
+}
+
 function _deleteNullVersionMD(bucketName, objKey, options, log, cb) {
-    metadata.deleteObjectMD(bucketName, objKey, options, log, err => {
-        if (err) {
-            log.debug('error from metadata deleting null version',
-            { error: err });
-        }
-        cb(err, options);
-    });
+    // before deleting null version md, retrieve location of data to delete
+    return _getVersionLocation(bucketName, objKey, options, log,
+        (err, nullDataToDelete) => {
+            if (err) {
+                log.warning('could not find null version metadata', {
+                    error: err,
+                    method: '_deleteNullVersionMD',
+                });
+                return cb(err);
+            }
+            return metadata.deleteObjectMD(bucketName, objKey, options, log,
```

Comment:
before this fix, we were deleting metadata, getting delete locations, then storing new metadata + deleting data. This presented a problem because the metadata was deleted before getting the data delete locations from it.